include/alpaka/mem/buf/SetKernel.hpp

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2020 Jeffrey Kelling
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/mem/buf/Traits.hpp>
+#include <alpaka/idx/Traits.hpp>
+#include <alpaka/idx/Accessors.hpp>
+#include <alpaka/idx/MapIdx.hpp>
+#include <alpaka/meta/Fold.hpp>
+
+namespace alpaka
+{
+    namespace mem
+    {
+        namespace view
+        {
+            //#############################################################################
+            //! any device ND memory set kernel.
+            class MemSetKernel
+            {
+            public:
+                //-----------------------------------------------------------------------------
+                //! The kernel entry point.
+                //!
+                //! All but the last element of threadElemExtent must be one.
+                //!
+                //! \tparam TAcc The accelerator environment to be executed on.
+                //! \tparam TExtent extent type.
+                //! \param acc The accelerator to be executed on.
+                //! \param val value to set.
+                //! \param dst target mem ptr.
+                //! \param extent area to set.
+                ALPAKA_NO_HOST_ACC_WARNING
+                template<
+                    typename TAcc,
+                    typename TExtent,
+                    typename TPitch>
+                ALPAKA_FN_ACC auto operator()(
+                    TAcc const & acc,
+                    std::uint8_t const val,
+                    std::uint8_t * dst,
+                    TExtent extent,
+                    TPitch pitch) const
+                -> void
+                {
+                    using Idx = typename idx::traits::IdxType<TExtent>::type;
+                    auto const gridThreadIdx(alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc));
+                    auto const threadElemExtent(alpaka::workdiv::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc));
+                    auto const idxThreadFirstElem = idx::getIdxThreadFirstElem(acc, gridThreadIdx, threadElemExtent);
+                    auto idx = idx::mapIdxPitchBytes<1u, dim::Dim<TAcc>::value>(idxThreadFirstElem, pitch)[0];
```

Comment:
I find the name `idx` confusing now. Could this be renamed to `offset`?

Reply:
On the other hand we are comparing and adding other index related variables, so it might be ok.

Reply:
I prefer idx, because of the mapping from `gridThreadIdx`.